scripts/create_xcframeworks.sh

```diff
@@ -1,72 +1,145 @@
-xcodebuild archive \
--scheme "Themis (iOS)" \
--destination="iOS" \
--archivePath $output_dir/archives/ios.xcarchive \
--derivedDataPath $output_dir/iphoneos \
--sdk iphoneos \
-SKIP_INSTALL=NO \
-BUILD_LIBRARIES_FOR_DISTRIBUTION=YES
-
-# build the framework for iOS simulator
-xcodebuild archive \
--scheme "Themis (iOS)" \
--destination="iOS Simulator" \
--archivePath $output_dir/archives/iossimulator.xcarchive \
--derivedDataPath $output_dir/iphoneos \
--sdk iphonesimulator \
-SKIP_INSTALL=NO \
-BUILD_LIBRARIES_FOR_DISTRIBUTION=YES
-
-# build the framework for macOS
-xcodebuild archive \
--scheme "Themis (macOS)" \
--destination="macOS" \
--archivePath $output_dir/archives/macosx.xcarchive \
--derivedDataPath $output_dir/macosx \
--sdk macosx \
-SKIP_INSTALL=NO \
-BUILD_LIBRARIES_FOR_DISTRIBUTION=YES
-
-# gather separate frameworks into a single xcframework
-xcodebuild -create-xcframework \
--framework $output_dir/archives/ios.xcarchive/Products/Library/Frameworks/themis.framework \
--framework $output_dir/archives/iossimulator.xcarchive/Products/Library/Frameworks/themis.framework \
--framework $output_dir/archives/macosx.xcarchive/Products/Library/Frameworks/themis.framework \
--output $output_dir/themis.xcframework
-
-# deleting the artifacts
-rm -rf $output_dir/archives
-rm -rf $output_dir/iphoneos
-rm -rf $output_dir/macosx
-
-# zip the xcodeframework
-# SPM accepts binary targets only in zip format
-cd $output_dir
-zip -r themis.xcframework.zip themis.xcframework
-
-rm -rf themis.xcframework
-
-# calculate checksum from the directory with Package.swift
-# update the the checksum in Package.swift if that is a new release
-cd $project_dir
-echo "XCF Checksum:"
-swift package compute-checksum $output_dir/themis.xcframework.zip
\ No newline at end of file
+
+clopenssl_scheme_ios="Themis (iOS)"
+clopenssl_scheme_mac="Themis (macOS)"
+clopenssl_output_dir=$BUILD_PATH/xcf_output/CLOpenSSL
+
+boringssl_scheme_ios="Themis (iOS) - BoringSSL"
+boringssl_scheme_mac="Themis (macOS) - BoringSSL"
+boringssl_output_dir=$BUILD_PATH/xcf_output/BoringSSL
+
+if [[ (! -d Themis.xcodeproj) || (! -f Package.swift) ]]
+then
+    echo >&2 "Please launch scripts/create_xcframeworks.sh from Themis repository root"
+    exit 1
+fi
+
+help() {
+    eval "echo \"$(cat $0 | awk 'NR == 3, /^$/ { print substr($0, 3) }')\""
+}
+
+build_clopenssl=
+build_boringssl=
+
+while [[ $# -gt 0 ]]
+do
+    case "$1" in
+      -h|--help) help; exit;;
+      --with-clopenssl) build_clopenssl=yes; shift;;
+      --with-boringssl) build_boringssl=yes; shift;;
+      *)
+        echo >&2 "Unknown option: $1"
+        echo >&2
+        help
+        exit 1
+        ;;
+    esac
+done
+
+if [[ -z "$build_clopenssl" && -z "$build_boringssl" ]]
+then
+    build_clopenssl=yes
+fi
+
+# CLOpenSSL builds expect Carthage dependencies to be fetched.
+# If they don't seem to be here, do a favor and pull them now.
+if [[ -n "$build_clopenssl" && ! -d Carthage ]]
+then
+    carthage bootstrap
```

Comment:
Actually, using Carthage is not the only option. You can add `clopenssl` to `Themis.xcodeproj` with SPM instead. In a perfect world, I'd prefer SPM over Carthage. I think in the future we'll work on a script that allows switching between these two.